### PR TITLE
chore(package upgrades): Migrate to Prisma v.4

### PR DESCRIPTION
**Pull request by nina-py**
## Goal

Update Prisma packages & schema to the latest major version. Upgrade guide is here: https://www.prisma.io/docs/guides/upgrade-guides/upgrading-versions/upgrading-to-prisma-4#client-changes. Luckily, only minor updates to the schema were needed (the explicit `@unique` modifiers). 

- Also fixed deprecation notices from the `faker` package that were adding noise to integration tests

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -30,80 +18,67 @@ model ApprovedItemAuthor {
...
   createdBy       String               @db.VarChar(255)
   updatedAt       DateTime             @updatedAt
   updatedBy       String?              @db.VarChar(255)
-
-  // associated models
-  scheduledItems ScheduledItem[]
-
-  // indexes
-  @@unique([externalId])
-  @@unique([url])
```
Comment (nina-py) [inline prisma/schema.prisma:63]: These have moved to the field declarations after running the upgrade script.

File: prisma/schema.prisma [hunk 2]
```diff
@@ -30,80 +18,67 @@ model ApprovedItemAuthor {
...
+  externalId String   @unique @default(uuid()) @db.VarChar(255)
+  url        String   @unique @db.VarChar(500)
   title      String?  @db.VarChar(255)
-  topic      String?   @db.VarChar(255)
+  topic      String?  @db.VarChar(255)
   language   String?  @db.VarChar(2)
   publisher  String?  @db.VarChar(255)
-  // Can be multiple reasons. For the MVP, Snowplow and the frontend
-  // will be the only consumers of this data.
-  // Will be likely stored either as comma-separated values or JSON.
```
Comment (nina-py) [inline prisma/schema.prisma:78]: The migration script stripped out all comments - perhaps for the better? As some of them appear to be outdated

File: prisma/schema.prisma [hunk 3]
```diff
@@ -30,80 +18,67 @@ model ApprovedItemAuthor {
...
-  scheduledItems ScheduledItem[]
-
-  // indexes
-  @@unique([externalId])
-  @@unique([url])
+  prospectId      String?              @db.VarChar(255)
+  isTimeSensitive Boolean              @default(false)
+  source          CuratedItemSource?
+  authors         ApprovedItemAuthor[]
+  scheduledItems  ScheduledItem[]
```
Comment (nina-py) [inline prisma/schema.prisma:42]: The migration script has also moved fields around for... reasons? Should I put them back where they were to reduce the diff?

File: src/admin/resolvers/mutations/ScheduledItem.ts [hunk 4]
```diff
@@ -11,7 +11,7 @@ import {
   AuthenticationError,
   UserInputError,
 } from '@pocket-tools/apollo-utils';
-import { PrismaClientKnownRequestError } from '@prisma/client/runtime';
+import { PrismaClientKnownRequestError } from '@prisma/client/runtime/library';
```
Comment (nina-py) [inline src/admin/resolvers/mutations/ScheduledItem.ts:14]: This also generated a "deprecated" notice in the tests

File: prisma/schema.prisma (hunk 3 above)
Comment (sri-kirsh) [reply prisma/schema.prisma:42]: ya looks like it grouped all fields that non primitive object types together

likewise, nullable fields at the end and non-nullable at start..this format is good, more organized.

**Review by sri-kirsh — APPROVED** [review]

**Pull request merged**